packages/flutter/test/material/carousel_test.dart

```diff
@@ -1193,6 +1193,55 @@ void main() {
     // No exception.
     expect(tester.takeException(), isNull);
   });
+
+  testWidgets('The shrinkExtent should keep the same when the item is tapped', (WidgetTester tester) async {
+    int offset = 0;
+
+    await tester.pumpWidget(
+      StatefulBuilder(
+        builder: (BuildContext context, StateSetter setState) {
+          return MaterialApp(
+            home: Scaffold(
+              body: Center(
+                child: ConstrainedBox(
+                  constraints: const BoxConstraints(maxHeight: 200),
+                  child: CarouselView(
+                    itemExtent: 330,
+                    onTap: (int idx) => setState(() {
+                      offset = idx;
+                    }),
+                    children: List<Widget>.generate(20, (int index) {
+                      index += offset;
+                      return ColoredBox(
+                        color: Colors.primaries[index % Colors.primaries.length].withOpacity(0.5),
+                        child: Center(
+                          child: Text('Item $index'),
+                        ),
+                      );
+                    }),
+                  ),
+                ),
+              ),
+            ),
+          );
+        }
+      )
+    );
```

Comment:
We can make the test a little bit simpler:)
```suggestion
  final List<Widget> children = List<Widget>.generate(20, (int index) {
      return Center(
        child: Text('Item $index'),
      );
    })
    await tester.pumpWidget(
      StatefulBuilder(
        builder: (BuildContext context, StateSetter setState) {
          return MaterialApp(
            home: Scaffold(
              body: Center(
                child: ConstrainedBox(
                  constraints: const BoxConstraints(maxHeight: 200),
                  child: CarouselView(
                    itemExtent: 330,
                    onTap: (int idx) => setState(() { }),
                    children: children,
                  ),
                ),
              ),
            ),
          );
        }
      )
    );
```